test/256_dns_persist_2_test.sh

```diff
@@ -0,0 +1,97 @@
+#! /bin/bash
+
+. ./config.sh
+
+wait_for_container() {
+    host="$1"
+    container="$2"
+    for i in $(seq 1 120); do
+        echo "Waiting for container $2 to start"
+        state=$(docker_on $host inspect -f "{{.State.Status}}" $container 2>/dev/null)
+        if [ "$state" == "running" ]; then
+            return
+        fi
+        sleep 1
+    done
+    echo "Timed out waiting for container $2 to start" >&2
+    exit 1
+}
+
+wait_for_dns_a_record() {
+    exp_name=${5:-$3}
+    for i in $(seq 1 30); do
+        echo "Waiting for $4 $exp_name"
+        actual=$(exec_on $1 $2 getent hosts $3 | tr -s ' ' | cut -d ' ' -f 1,2)
+        if [ "$actual" == "$4 $exp_name" ]; then
+            return
+        fi
+        sleep 1
+    done
+    echo "Timed out waiting for $4 $exp_name" >& 2
+    exit 1
+}
+
+C1=10.2.0.78
+C2=10.2.0.34
+IP=192.168.32.25
+NAME=seetwo.weave.local
+NAME1=seetwoextra.weave.local
+NAME2=extern.weave.local
+
+start_suite "Check persisted DNS records"
+
+weave_on $HOST1 launch
+weave_on $HOST2 launch $HOST1
+
+CMD="docker run --restart always --name c2 -h $NAME -e WEAVE_CIDR=$C2/24 -td $SMALL_IMAGE /bin/sh"
+run_on $HOST1 "eval \$(weave env) ; $CMD"
+start_container_with_dns $HOST1 $C1/24 --name=c1
+start_container_with_dns $HOST2 --name=c3
+weave_on $HOST1 dns-add c2 -h $NAME1
+weave_on $HOST1 dns-add $IP -h $NAME2
+
+assert_dns_a_record $HOST1 c1 $NAME $C2
+assert_dns_a_record $HOST1 c1 $NAME1 $C2
+assert_dns_a_record $HOST1 c1 $NAME2 $IP
```

Comment:
there seem to be repeated groups of 3 commands, which should be pulled out to a function